server/src/main/java/org/opensearch/extensions/ExtensionRequest.java

```diff
@@ -26,54 +26,54 @@
  */
 public class ExtensionRequest extends TransportRequest {
     private static final Logger logger = LogManager.getLogger(ExtensionRequest.class);
-    private final ExtensionsManager.RequestType requestType;
-    private final Optional<String> uniqueId;
+    private final ExtensionRequestProto.ExtensionRequest request;
 
-    public ExtensionRequest(ExtensionsManager.RequestType requestType) {
+    public ExtensionRequest(ExtensionRequestProto.RequestType requestType) {
         this(requestType, null);
     }
 
-    public ExtensionRequest(ExtensionsManager.RequestType requestType, @Nullable String uniqueId) {
-        this.requestType = requestType;
-        this.uniqueId = uniqueId == null ? Optional.empty() : Optional.of(uniqueId);
+    public ExtensionRequest(ExtensionRequestProto.RequestType requestType, @Nullable String uniqueId) {
+        ExtensionRequestProto.ExtensionRequest.Builder builder = ExtensionRequestProto.ExtensionRequest.newBuilder();
+        if (uniqueId != null) {
```

Comment:
How are we handling the case when `uniqueID` is null?

Reply:
We dont have to. Protobuf already sets default values[1].

[1] https://protobuf.dev/programming-guides/proto3/#default